src/net/sourceforge/kolmafia/swingui/widget/EditableAutoFilterComboBox.java

```diff
@@ -0,0 +1,77 @@
+package net.sourceforge.kolmafia.swingui.widget;
+
+import java.awt.event.KeyEvent;
+import net.java.dev.spellcast.utilities.LockableListModel;
+
+public class EditableAutoFilterComboBox extends AutoFilterComboBox<String> {
+  public EditableAutoFilterComboBox(final LockableListModel<String> model) {
+    super(model);
+  }
+
+  public String getText() {
+    return (String) (this.getSelectedItem() != null ? this.getSelectedItem() : this.currentMatch);
```

Comment:
(I hate that LcokableListModel.getSelectedItem returns an Object instead of a genericized type...)

Reply:
(why are we even still using that? Wasn't it used for the lockable aspect, which got removed?)

Reply:
At this point, LockableListModel is a (Swing) ListModel that has a notion of visible / hidden elements based on a custom filter.